internal/encryption/pkcs11.go

```diff
@@ -0,0 +1,194 @@
+package encryption
+
+import (
+	"crypto/rand"
+	"fmt"
+
+	"github.com/miekg/pkcs11"
+)
+
+// PKCS11Backend implements EncryptionBackend using PKCS11 protocol for HSMs.
+type PKCS11Backend struct {
+	ctx   *pkcs11.Ctx
+	pin   string
+	keyID uint16
+}
+
+const ivSize = 16
```

Comment:
Since it's in bytes, I would recommend adding a `// bytes` next to the value to make it clear to developers what we are talking about. Or even add bytes to the variable name.